### PR TITLE
add sth mongodb indexes

### PR DESCRIPTION
**Pull request by AlvaroVega**
fix issue https://github.com/telefonicaid/orchestrator/issues/192

### CODE REVIEW

File: src/orchestrator/core/mongo.py [hunk 1]
```diff
@@ -58,6 +59,20 @@ def createIndexes(self, SERVICE_NAME):
         except Exception, e:
             logger.warn("createIndex database %s exception: %s" % (databaseName,e))
 
+
+    def createSTHIndexes(self, SERVICE_NAME, SUBSERVICE_NAME):
+        try:
+            # For STH
+            databaseName = 'sth_' + SERVICE_NAME
+            db = self.client[databaseName]
+            collectionName = 'sth_/' + SUBSERVICE_NAME
```
Comment (fgalan) [inline src/orchestrator/core/mongo.py]: Maybe it should be  `collectionName = 'sth_/' + SUBSERVICE_NAME + '.aggr'` ?

In addition, it seems the code is creating four one-key indexes. However, what should be done is create one four-keys index.

Comment (AlvaroVega) [reply src/orchestrator/core/mongo.py]: Both fixed in https://github.com/telefonicaid/orchestrator/pull/204/commits/1507dbcd7bc44d714e34a5bb6d3741d9c32880c7

Comment (AlvaroVega) [reply src/orchestrator/core/mongo.py]: One-key indexes is good for search by all fields, like `index _id.entityId, _id.attrName, _id.resolution, _id.origin` but I'm not sure if it's optimized when you search just by entityId of attrName. Are we sure that the most common query search included all that fields?

Comment (fgalan) [reply src/orchestrator/core/mongo.py]: As far as I checked using MongoDB profiler, the query (applied in a $match statement in an aggregation pipeline is this):

```
                                "$match" : {
                                        "_id.entityId" : "NativeModel.50043034",
                                        "_id.entityType" : "device",
                                        "_id.attrName" : "batteryCharge",
                                        "_id.resolution" : "hour",
                                        "_id.origin" : {
                                                "$lte" : ISODate("2019-10-26T00:00:00Z"),
                                                "$gte" : ISODate("2018-10-16T00:00:00Z")
                                        }
                                }
```

so I'm pretty sure.

File: src/orchestrator/core/mongo.py [hunk 2]
```diff
@@ -58,6 +59,20 @@ def createIndexes(self, SERVICE_NAME):
         except Exception, e:
             logger.warn("createIndex database %s exception: %s" % (databaseName,e))
 
+
+    def createSTHIndexes(self, SERVICE_NAME, SUBSERVICE_NAME):
+        try:
+            # For STH
```
Comment (fgalan) [inline src/orchestrator/core/mongo.py]: I understand the index for raw collection should be also created, in a similar way. In this case the index structure is:

```
{entityId: 1, entityType: 1, attrName: 1, recvTime: -1}
```

In addition, I'd suggest to include a link in this part to https://github.com/telefonicaid/fiware-sth-comet/blob/master/doc/manuals/db_indexes.md, for reference to know where these indexes come.

File: src/orchestrator/core/mongo.py (hunk 1 above)
Comment (AlvaroVega) [reply src/orchestrator/core/mongo.py]: Ok. NTC

File: src/orchestrator/core/mongo.py (hunk 2 above)
Comment (AlvaroVega) [reply src/orchestrator/core/mongo.py]: fixed in https://github.com/telefonicaid/orchestrator/pull/204

**Comment by fgalan** [comment]
I have added a couple of final fixes in 9c16327aa412f0fb9dd2d1cee033ed1f603755e1. Please have a look to check if you think they are fine.

**Comment by AlvaroVega** [comment]
LGTM

**Pull request merged**